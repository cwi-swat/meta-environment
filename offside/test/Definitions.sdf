module Definitions

exports

sorts Program

hiddens
sorts Id Exp Definition

exports
lexical syntax

[A-Za-z0-9]+ -> Id
"definition" -> Id {reject}

[\ \t\n] -> LAYOUT

lexical restrictions
  Id -/- [A-Za-z0-9]

context-free restrictions
 LAYOUT? -/- [\ \t\n]

context-free syntax

"program"  Definition+ -> Program
"definition" Exp+      -> Definition {offside}

Definition -> Exp 
Id         -> Exp
  
  father
    term ;
  father term;

  [father
    term;]
    [father term]

  [father
    term;
    [father term]]
