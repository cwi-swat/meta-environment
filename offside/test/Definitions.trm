program

 definition
  a
definition
  b
