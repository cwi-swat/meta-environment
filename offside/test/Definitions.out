parsetree ( appl ( prod ( [ cf ( opt ( layout ) ) , cf ( sort ( "Program" ) ) , cf ( opt ( layout ) ) ] , sort ( "<START>" ) , no-attrs ) , [ appl ( prod ( [  ] , cf ( opt ( layout ) ) , no-attrs ) , [  ] ) , appl ( prod ( [ lit ( "program" ) , cf ( opt ( layout ) ) , cf ( iter ( sort ( "Definition" ) ) ) ] , cf ( sort ( "Program" ) ) , no-attrs ) , [ lit ( "program" ) , appl ( prod ( [ cf ( layout ) ] , cf ( opt ( layout ) ) , no-attrs ) , [ 10 , 10 , 32 ] ) , appl ( list ( cf ( iter ( sort ( "Definition" ) ) ) ) , [ appl ( prod ( [ lit ( "definition" ) , cf ( opt ( layout ) ) , cf ( iter ( sort ( "Exp" ) ) ) ] , cf ( sort ( "Definition" ) ) , attrs ( [ term ( offside ) ] ) ) , [ lit ( "definition" ) , appl ( prod ( [ cf ( layout ) ] , cf ( opt ( layout ) ) , no-attrs ) , [ 10 , 32 , 32 ] ) , appl ( list ( cf ( iter ( sort ( "Exp" ) ) ) ) , [ appl ( prod ( [ cf ( sort ( "Id" ) ) ] , cf ( sort ( "Exp" ) ) , no-attrs ) , [ appl ( prod ( [ lex ( sort ( "Id" ) ) ] , cf ( sort ( "Id" ) ) , no-attrs ) , [ appl ( list ( iter-star ( char-class ( [ range ( 0 , 255 ) ] ) ) ) , [ 97 ] ) { [ pos-info , area ( "offside" , 4 , 2 , 4 , 3 ) ] } ] ) { [ pos-info , area ( "offside" , 4 , 2 , 4 , 3 ) ] } ] ) { [ pos-info , area ( "offside" , 4 , 2 , 4 , 3 ) ] } ] ) { [ pos-info , area ( "offside" , 4 , 2 , 4 , 3 ) ] } ] ) { [ pos-info , area ( "offside" , 3 , 1 , 4 , 3 ) ] } , appl ( prod ( [ cf ( layout ) ] , cf ( opt ( layout ) ) , no-attrs ) , [ 10 ] ) , appl ( prod ( [ lit ( "definition" ) , cf ( opt ( layout ) ) , cf ( iter ( sort ( "Exp" ) ) ) ] , cf ( sort ( "Definition" ) ) , attrs ( [ term ( offside ) ] ) ) , [ lit ( "definition" ) , appl ( prod ( [ cf ( layout ) ] , cf ( opt ( layout ) ) , no-attrs ) , [ 10 , 32 , 32 ] ) , appl ( list ( cf ( iter ( sort ( "Exp" ) ) ) ) , [ appl ( prod ( [ cf ( sort ( "Id" ) ) ] , cf ( sort ( "Exp" ) ) , no-attrs ) , [ appl ( prod ( [ lex ( sort ( "Id" ) ) ] , cf ( sort ( "Id" ) ) , no-attrs ) , [ appl ( list ( iter-star ( char-class ( [ range ( 0 , 255 ) ] ) ) ) , [ 98 ] ) { [ pos-info , area ( "offside" , 6 , 2 , 6 , 3 ) ] } ] ) { [ pos-info , area ( "offside" , 6 , 2 , 6 , 3 ) ] } ] ) { [ pos-info , area ( "offside" , 6 , 2 , 6 , 3 ) ] } ] ) { [ pos-info , area ( "offside" , 6 , 2 , 6 , 3 ) ] } ] ) { [ pos-info , area ( "offside" , 5 , 0 , 6 , 3 ) ] } ] ) { [ pos-info , area ( "offside" , 3 , 1 , 6 , 3 ) ] } ] ) { [ pos-info , area ( "offside" , 1 , 0 , 6 , 3 ) ] } , appl ( prod ( [ cf ( layout ) ] , cf ( opt ( layout ) ) , no-attrs ) , [ 10 ] ) ] ) { [ pos-info , area ( "offside" , 1 , 0 , 7 , 0 ) ] } , 0 )
